### ISSUE TITLE
Support CommonMark syntax

### ISSUE THREAD

**Issue by xgouchet** [retext-project/retext#253]
Currently ReText gives the option to use Markdown, Textile and reStructured Text. It would be great to be compliant with CommonMark (which is a kind of standardized version of Markdown)

**Comment by mitya57** [comment]
Out of curiosity: what the the syntax features that are present in CommonMark that you miss with the current implementation? I am quite sure there are Python-Markdown extensions that implement the syntax you need.

That said, I can add the CommonMark markup, but I am not going to reassign the standard file extensions (`.mkd`, `.md`, `.markdown`) to it, so all current files will still load using Python-Markdown.

**Comment by xgouchet** [comment]
It's not so much mssing syntax feature, rather than having a standard syntax across several systems. 

There are lots of different markdown flavor, based on website and parsers, and CommonMark actually have a spec on how to write syntax. 

I understand the problem with the file extension, maybe you could add a user preference to choose between python markdown / common mark ?

**Comment by keesse** [comment]
I agree with xgouchet, that would be nice, a lot of other applications also support both.
E.g. a '\' instead of '  ' at the end of a line is more visable (allthough ReText prints grey spaces, not all other editers do).

**Comment by kwill** [comment]
Some additional comments (just putting these links here for reference):

* ReText currently relies on Python-Markdown which will never support CommonMark (see https://github.com/Python-Markdown/markdown/issues/338 and https://github.com/Python-Markdown/markdown/issues/545 )
* [CommonMark-py](https://github.com/rtfd/CommonMark-py) is a Python implementation of CommonMark; https://github.com/retext-project/pymarkups/pull/2 is an incomplete attempt to provide it as an alternate backend

**Comment by matkoniecz** [comment]
> Out of curiosity: what the the syntax features that are present in CommonMark that you miss with the current implementation? I am quite sure there are Python-Markdown extensions that implement the syntax you need.

In some cases CommonsMark and ReText gives different rendering, so I need to make changes like https://github.com/colouring-london/colouring-london/commit/b07de223b124fa18cd2373294b627c51edfb9d7f just because ReText is displaying it brokenly.

Specifically:

```
This repository will contain open-source code for the project which:
- stores building footprint polygons and source metadata
- allows site users to record building attribute data
```

is shown as list on GitHub that is using CommonsMark and not in ReText that is using MarkDown.

https://johnmacfarlane.net/babelmark2/?text=This+repository+will+contain+open-source+code+for+the+project+which%3A%0A-+stores+building+footprint+polygons+and+source+metadata%0A-+allows+site+users+to+record+building+attribute+data

![screen](https://user-images.githubusercontent.com/899988/140515612-7eda873c-1011-468f-90b5-c8a083021459.png)

**Comment by mitya57** [comment]
As I said earlier,

> I am quite sure there are Python-Markdown extensions that implement the syntax you need.

And here is an extension for your case:

https://github.com/adamb70/mdx-breakless-lists

**Comment by matkoniecz** [comment]
@mitya57 How one may use them in ReText? I see no obvious way to achieve this.

**Comment by mitya57** [comment]
There are multiple ways to enable extensions, as described in the wiki: https://github.com/retext-project/retext/wiki/Markdown-extensions.

- Directly in document by adding `<!-- Required extensions: mdx_breakless_lists -->` as first line.
- `markdown-extensions.txt` or `markdown-extensions.yaml` in the document directory.
- `markdown-extensions.txt` or `markdown-extensions.yaml` in your user `~/.config`.

And of course you need to install the extension, in this case `pip3 install mdx-breakless-lists`.

![Screenshot](https://user-images.githubusercontent.com/1381584/140558953-06f30fe5-1a8f-4a7f-b349-1c88e40310ad.png)
